Library/Homebrew/formula_installer.rb

```diff
@@ -79,17 +79,31 @@ def pour_bottle?(install_bottle_options = { :warn=>false })
     return false if @pour_failed
 
     bottle = formula.bottle
-    return true  if force_bottle? && bottle
+    return false unless bottle
+    return true  if force_bottle?
     return false if build_from_source? || build_bottle? || interactive?
     return false if ARGV.cc
     return false unless options.empty?
     return false if formula.bottle_disabled?
     return true  if formula.local_bottle_path
-    return false unless bottle && formula.pour_bottle?
+    return false unless formula.pour_bottle?
+
+    if formula.pour_bottle_block && !formula.pour_bottle_block.call
+      if install_bottle_options[:warn] && formula.pour_bottle_reason
+        opoo <<-EOS.undent
+          Building #{formula.full_name} from source:
+            #{formula.pour_bottle_reason}
+        EOS
+      end
+      return false
+    end
 
     unless bottle.compatible_cellar?
       if install_bottle_options[:warn]
-        opoo "Building source; cellar of #{formula.full_name}'s bottle is #{bottle.cellar}"
+        opoo <<-EOS.undent
+          Building #{formula.full_name} from source:
+            The bottle needs a #{bottle.cellar} cellar (yours is #{HOMEBREW_CELLAR}).
```

Comment:
I think in this context "Cellar" is a place, in the same way "London" is a place, and thus needs capitalised. That was the point behind my nit, really. Not a major thing though.